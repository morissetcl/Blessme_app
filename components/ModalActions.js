import React, { Component } from 'react';
import { StyleSheet, View, TouchableOpacity, Alert } from 'react-native';
import { destroyPrayerResquest } from '../api/PrayerRequest';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faEllipsisV } from '@fortawesome/free-solid-svg-icons';
import { displayMessage } from "./shared/message";

export default class ModalActions extends Component {
  constructor(props) {
    super(props);
    this.state = {
      prayerRequest: props.prayerRequest,
      navigation: props.navigation
    };
  }

  _deletePrayerRequest = () => {
    destroyPrayerResquest({
      prayerRequestId: this.state.prayerRequest.id,
      navigation: this.state.navigation }).then(() => {
        displayMessage('Votre demande a bien été supprimée.', 'success')
    });
  }

  _goToPrayerRequestForm = () => {
    this.state.navigation.navigate('PrayerRequestForm', { currentUserEmail: this.state.currentUserEmail,  prayerRequest: this.state.prayerRequest, editPrayer: true });
  }

  _showAlert = () => {
    Alert.alert(
      this.state.title,
      'Que voulez vous faire avec cette demande ?',
      [
        {text: 'Modifier', onPress: () => this._goToPrayerRequestForm()},
        {text: '', onPress: () => alert('')},
        {text: 'Supprimer', onPress: () => this._deletePrayerRequest()}
      ],
      { onDismiss: () => {} }
    )
  }

  render() {
    return (
      <TouchableOpacity
        onPress={this._showAlert}
        style = {styles.menu} >
        <FontAwesomeIcon icon={ faEllipsisV } size={16} color={ '#bbbbbb' }/>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  menu: {
    position: 'absolute',
    top: 8,
    right: 0,
  }
});
