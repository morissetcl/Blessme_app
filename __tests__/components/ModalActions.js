import React from 'react'
import { shallow, mount, render } from 'enzyme';
import ModalActions from '../../components/ModalActions'
import { Provider } from 'react-redux';

import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});


jest.mock('../../api/GetApiUrl.js', () => ({
  MY_CONSTANT: 'something fake'
}));

jest.mock('react-native', () => {
  return {
    StyleSheet: {
      create: () => ({}),
    },
  };
});

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: ''
}))
jest.mock('react-native-flash-message', () => ({
    FontAwesomeIcon: ''
}))
jest.mock('expo-localization', () => ({
    FontAwesomeIcon: ''
}))
jest.mock('react-native-material-menu', () => ({
    FontAwesomeIcon: ''
}))

describe('ModalActions', () => {
  let wrapper;

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have the correct props', () => {
    const wrapper = shallow(<Provider store={store} signal={true}><ModalActions /></Provider>);
    console.log(wrapper.dive())

  });
})
