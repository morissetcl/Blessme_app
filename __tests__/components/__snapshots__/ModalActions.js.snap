// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalActions Should render. 1`] = `undefined`;
